compiler/java/src/main/java/com/c/lexer/Lexer.java

```diff
@@ -1,29 +1,635 @@
+        StringBuilder value = new StringBuilder();
+        while (!isLookAheadCommentLineIdentifier() && !isLookAheadCommentBlockOpenIdentifier() && !isEOF()) {
+            value.append(reader.lookAhead());
+            reader.advance();
+        }
+        currentContext = Context.SECTION;
+        return constructToken(TokenType.ARBITRARY, value.toString());
+    }
+
+    /**
+     * Consumes the payload that might get uncommented if the respective conditional comment evaluates to true.
+     *
+     * @return a Token of type ARBITRARY and the char sequence that might get uncommented as value of that Token
+     */
+    private Token consumePayload() {
+        StringBuilder value = new StringBuilder();
+        while (!isLookAheadCommentLineIdentifier()
+                && !isLookAheadCommentBlockCloseIdentifierWithBrace()
+                && !isEOF()) {
+
+            // Ignore payload comment identifiers
+            if (isLookAheadCommentPayloadIdentifier()) {
+                for (int i = 0; i < commentPayloadIdentifier.length(); i++) {
+                    reader.advance();
+                }
+            }
+
+            // TODO: also allow for any chars after '}' and before '-->' (HTML)
+            // However, upon first evaluation, this would long for an infinite look ahead
+
+            value.append(reader.lookAhead());
+            reader.advance();
+        }
+
+        currentContext = Context.SECTION;
+        return constructToken(TokenType.ARBITRARY, value.toString());
+    }
+
+    /**
+     * Consumes part of a conditional comments section. One section can consist of multiple conditional comments,
+     * e.g. multiple if statements. A conditional comments section is supposed to include a payload that might get
+     * uncommented if the respective conditional comment evaluates to true.
+     * <p>
+     * Note that consumeSection() does NOT consume the whole section at once but step-by-step (with multiple calls to
+     * this method). It constructs exactly one new token with every call to this method.
+     *
+     * @return a Token describing part of the conditional comments section, e.g. an "if" identifier
+     * @throws UnexpectedCharException if a char - read by the FileReader - was not the expected one
+     * @throws UnknownCharException    if a char cannot be processed to a valid Token
+     */
+    private Token consumeSection() throws UnexpectedCharException, UnknownCharException {
+        // Check for Tokens first that require more than one character look ahead (!)
+        if (isLookAheadCommentLineIdentifier()) return consumeCommentLineIdentifier();
+        if (isLookAheadCommentBlockOpenIdentifier()) return consumeCommentBlockOpenIdentifier();
+        if (isLookAheadCommentBlockCloseIdentifier()) return consumeCommentBlockCloseIdentifier();
+        if (isLookAheadCommentPayloadIdentifier()) return consumeCommentPayloadIdentifier();
+
+        switch (reader.lookAhead()) {
```

Comment:
Outsourced this to a local variable since it looks nicer. But I don't think that this yields any significant performance increase since `reader.lookAhead()` is simply returning a local variable itself. But yeah, there's another stack frame for calling the method ;)